tools/clang/unittests/HLSLExec/LongVectorOps.def

```diff
@@ -0,0 +1,109 @@
+//
+//
+//
+
+#ifndef INPUT_SET
+#define INPUT_SET(SYMBOL)
+#endif
+
+INPUT_SET(Default1)
+INPUT_SET(Default2)
+INPUT_SET(Default3)
+INPUT_SET(SmoothStepMin)
+INPUT_SET(SmoothStepMax)
+INPUT_SET(SmoothStepInput)
+INPUT_SET(RangeOne)
+INPUT_SET(RangeHalfPi)
+INPUT_SET(SplitDouble)
+INPUT_SET(BitShiftRhs)
+
+#undef INPUT_SET
+
+#ifndef OP
+#define OP(GROUP, SYMBOL, ARITY, INTRINSIC, OPERATOR, DEFINES, INPUT_SET_1,    \
+           INPUT_SET_2, INPUT_SET_3)
+#endif
+
+#define OP_DEFAULT_DEFINES(GROUP, SYMBOL, ARITY, INTRINSIC, OPERATOR, DEFINES) \
+  OP(GROUP, SYMBOL, ARITY, INTRINSIC, OPERATOR, DEFINES, Default1, Default2,   \
+     Default3)
+
+#define OP_DEFAULT(GROUP, SYMBOL, ARITY, INTRINSIC, OPERATOR)                  \
+  OP_DEFAULT_DEFINES(GROUP, SYMBOL, ARITY, INTRINSIC, OPERATOR, "")
+
+OP_DEFAULT(TernaryMath, Mad, 3, "mad", "")
+OP(TernaryMath, SmoothStep, 3, "smoothstep", "", "", SmoothStepMin,
+   SmoothStepMax, SmoothStepInput)
+OP_DEFAULT(TernaryMath, Fma, 3, "fma", "")
+
+OP_DEFAULT(BinaryMath, Add, 2, "", "+")
```

Comment:
nit: Wondering if 'UNARY_OP', 'BINARY_OP', and 'TERNARY_OP' would improve readability. And removes the need to pass a value for ARITY here.